pkg/backup/http.go

```diff
@@ -69,12 +71,37 @@ func (b *Backup) serveSnap(w http.ResponseWriter, r *http.Request) {
 		http.NotFound(w, r)
 		return
 	}
+
+	versionValue := r.FormValue(constants.BackupParamEtcdVersion)
+	reqV, err := getComparableEtcdVersion(strings.TrimLeft(versionValue, "v"))
+	if err != nil {
+		http.Error(w, fmt.Sprintf("invalid param 'version' (%s): %v", versionValue, err), http.StatusBadRequest)
+		return
+	}
+	serV, err := getEtcdVersionFromSnap(fname)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("fail to parse etcd version from file (%s): %v", fname, err), http.StatusInternalServerError)
+		return
+	}
+	if _, ok := validEtcdVersionsForBackup[reqV][serV]; !ok {
```

Comment:
we need to check if validEtcdVersionsForBackup[reqV] is nil. or this line might panic the backup tool.